Smart Train Observer

curryducker opened the pull request.

curryducker committed:
Added Smart Train Observer

- Smart Train Observers are able to filter train according to a set condition.
- Conditions can be set in the GUI.
- Mod developers can add more conditions using the registry.
- Comes with 6 built-in conditions.

TODO:
- Ponder Scene
- Invert Condition Option

curryducker committed:
Fix Loading Order

curryducker committed:
Add inverted conditions

curryducker committed:
Add Smart Train Observer Ponder Scene

curryducker committed:
Bump mod version

phantomeye1931 commented:
Ik wil best reviewen maar zal even moeten wachten tot ik in nederland ben... rond 18 aug op zn laatst

LarsMans64 commented on src/main/java/nl/teamdiopside/expandingtechnologies/blocks/observer/SmartTrainObserver.java:
Comments needed

LarsMans64 commented on src/main/java/nl/teamdiopside/expandingtechnologies/mixin/TrainMixin.java:
weird braces

LarsMans64 commented on src/main/resources/assets/expandingtechnologies/blockstates/smart_train_observer.json:
uvlock

LarsMans64 commented on src/main/resources/assets/expandingtechnologies/lang/default/interface.json:
current destination -> next station

LarsMans64 commented on src/main/resources/assets/expandingtechnologies/lang/en_us.json:
current destination -> next station

curryducker committed:
Remove weird braces in Mixins

curryducker committed:
Cleanup

- Remove unnecessary UV lock in the Smart Train Observer blockstate
- Add comments to SmartTrainObserver
- Clarify condition language keys

curryducker committed:
Fix suggestions on servers and bump mod version
